### DISCUSSION TITLE
How to fix an incorrect node name?

### DISCUSSION THREAD

**Discussion by abesinger** [Security-Onion-Solutions/securityonion#6437] [Unsupported Versions] [answered]
I just added a search node to a distributed install (2.3.80), and then discovered that I'd fat-fingered the hostname, so now `hostname` doesn't match DNS (perhaps that's an installer bug/oversight?).  How do I properly correct the node's hostname?  What I've gathered so far is:

- change name in `/opt/so/saltstack/local/pillar/data/nodestab.sls`
- rename `/opt/so/saltstack/local/pillar/<mangled>_searchnode.sls`
- change the hostname in the following files on the node itself:
  - `/etc/hostname`
  - `/etc/hosts`
  - `/etc/salt/minion`
- reboot minion, reboot manager (perhaps not necessary, but what the heck)

Am I missing anything?  I see on the search node that there are multiple references to the hostname in `/opt/so/conf/*`, but I assume those will get updated by Salt.

**Comment by dougburks** [comment, marked as answer]
I'm not sure that we've ever tried to do something like this so we can't provide any guarantees that it will work for you. You might try testing on a test deployment, perhaps with VM snapshots in case you need to revert.

**Reply by abesinger** [reply]
I definitely can test in a VM, but before I do, you didn't quite answer my question: did I miss anything, or in other words, is there anything else you can think of that I might need to change?

To create the list above,  I grepped for the hostname in `/opt/so/saltstack` on the manager, and` /etc` on the misnamed node. Anyplace else?  Perhaps embedded in a container (though I _think_ SO generates those with Salt) or in an SSL cert, or ??

**Reply by m0duspwnens** [reply]
On the remote node, you will need to remove `minion.pem` and `minion.pub` from `/etc/salt/pki/minion/`. Then you will need to delete the the old Salt key from the manager with `salt-key -d <minionid>`.  After that is done, you will need to run `salt-call state.show_top` on the remote node, then accept the key on the manager with `salt-key -a <minionid>`. After you have modified `/etc/salt/minion` on the remote node, you will also want to restart the `salt-minion` service as well.

This is the only other thing that I can think of that would need to be done. As Doug mentioned, we haven't tested this before, so the exact steps are tough to provide.

**Reply by abesinger** [reply]
that's very helpful, thanls. I'll let you know how it goes.
